reference/src/client/test/sa_key_provision_netflix.cpp

```diff
@@ -0,0 +1,475 @@
+        INFO("ESN length : %d", nflxProvision->esnContainerLength);
+
+        status = sa_key_provision_ta(NETFLIX_PROVISIONING, nflxProvision,
+           sizeof(NetflixProvisioning), nullptr);
+        ASSERT_EQ(status, SA_STATUS_OK);
+
+        if(nullptr != nflxProvision) {
+           delete nflxProvision;
+           nflxProvision = nullptr;
+        }
+    }
+    TEST_F(SaKeyProvisionNetflixTest, failsZeroInLength) {
+        auto key_length = SYM_128_KEY_SIZE;
+        std::vector<uint8_t> clear_encryption_key = random(key_length);
+        std::vector<uint8_t> clear_hmac_key = random(key_length);
+
+        //create a hmac key container
+        std::string hmac_key_container;
+        auto hmac_key_type = SA_KEY_TYPE_SYMMETRIC;
+        std::string hmac_key_type_string = "HMAC-128";
+        sa_status status = create_key_container(
+             hmac_key_type_string, //key type string
+             hmac_key_type, //key type
+             clear_hmac_key,
+             hmac_key_container);
+        ASSERT_EQ(SA_STATUS_OK, status);
+        INFO("hmac_key_container length : %d", hmac_key_container.size());
+
+        //create a wrapping key container
+        std::string wrapping_key_container;
+        auto wrapping_key_type = SA_KEY_TYPE_SYMMETRIC;
+        std::string wrapping_key_type_string = "AES-128";
+        std::vector<uint8_t> key_id = random(AES_BLOCK_SIZE);
+        //You can use other algorithm to wrap key, here just provide
+        //a very simple way wrap AES key.
+        for (auto i = 0; i < AES_BLOCK_SIZE; i++) {
+            clear_encryption_key[i] ^= key_id[i];
+        }
+        status = create_key_container(
+             wrapping_key_type_string, //key type string
+             wrapping_key_type, //key type
+             clear_encryption_key,
+             wrapping_key_container);
+        ASSERT_EQ(SA_STATUS_OK, status);
+        INFO("wrapping_key_container length : %d", wrapping_key_container.size());
+
+        NetflixProvisioning *nflxProvision = new NetflixProvisioning;
+        ASSERT_NE(nullptr, nflxProvision);
+        nflxProvision->hmacKey = (void*)hmac_key_container.data();
+        nflxProvision->hmacKeyLength = hmac_key_container.size();
+        nflxProvision->wrappingKey = (void*)wrapping_key_container.data();
+        nflxProvision->wrappingKeyLength = wrapping_key_container.size();
+        nflxProvision->esnContainer = (void*)(random(SYM_256_KEY_SIZE)).data();
+        nflxProvision->esnContainerLength = SYM_256_KEY_SIZE;
+
+        INFO("hmac length : %d", nflxProvision->hmacKeyLength);
+        INFO("wrapping length : %d", nflxProvision->wrappingKeyLength);
+        INFO("ESN length : %d", nflxProvision->esnContainerLength);
+
+        status = sa_key_provision_ta(NETFLIX_PROVISIONING, nflxProvision,
+           0, nullptr);
+
+        if (nullptr != nflxProvision) {
+           delete nflxProvision;
+           nflxProvision = nullptr;
+        }
+        ASSERT_NE(status, SA_STATUS_OK);
```

Comment:
We should use a similar test strategy as other SecAPI3 tests where the failure that is being tested is asserted. Looks like this applies to all tests in this PR. Use other secAPI3 client tests as a guide such as those for sa_key_import.